src/vs/workbench/api/browser/mainThreadAuthentication.ts

```diff
@@ -536,4 +544,40 @@ export class MainThreadAuthentication extends Disposable implements MainThreadAu
 		}
 		return false;
 	}
+
+	async $promptForClientDetails(authorizationServerUrl: string): Promise<{ clientId: string; clientSecret?: string } | undefined> {
+		const clientId = await this.quickInputService.input({
+			title: nls.localize('clientIdPromptTitle', "OAuth Client ID Required"),
+			prompt: nls.localize('clientIdPrompt', "The authorization server '{0}' does not support automatic client registration. Please enter a client ID that has been registered for VS Code:", authorizationServerUrl),
+			placeHolder: nls.localize('clientIdPlaceholder', "Enter your OAuth client ID"),
+			ignoreFocusLost: true,
+			validateInput: (value: string) => {
+				if (!value || value.trim().length === 0) {
+					return nls.localize('clientIdRequired', "Client ID is required");
+				}
+				return undefined;
+			}
+		});
+
+		if (!clientId || clientId.trim().length === 0) {
+			return undefined;
+		}
+
+		const clientSecret = await this.quickInputService.input({
+			title: nls.localize('clientSecretPromptTitle', "OAuth Client Secret Required"),
```

Comment:
The title should be shared across these quick picks because this is essentially a multi-step quick pick. Make the title something like: "Add Client Registration Details"